core/client/src/main/java/alluxio/client/file/FileOutStream.java

```diff
@@ -135,12 +135,15 @@ public FileOutStream(AlluxioURI path, OutStreamOptions options, FileSystemContex
           mUfsFileId =
               mFileSystemWorkerClient.createUfsFile(new AlluxioURI(mUfsPath),
                   CreateUfsFileOptions.defaults().setPermission(perm));
+          mUnderStorageOutputStream = mUnderOutStreamFactory
+              .create(mFileSystemWorkerClient.getWorkerDataServerAddress(), mUfsFileId);
         } catch (AlluxioException e) {
           mFileSystemWorkerClient.close();
           throw new IOException(e);
+        } catch (IOException e) {
```

Comment:
but the code path is different. one convert `e` to a new `IOException` (and making `e` as cause), one throw `e` directly. I could convert `IOException` to `IOException` to consolidate the catch block, but I feel it's perhaps less ideal as `IOException` will lose information,